docker/alternator-dns/dns_server.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+import sys
+import dnslib.server
+import dnslib
+import random
+import _thread
+import urllib.request
+import time
+import socket
+
+from dnslib import DNSRecord, RCODE
+
+# The list of live nodes, all of them supposedly answering HTTP requests on
+# alternator_port. One of these nodes will be returned at random from every
+# DNS request. This list starts with one or more known nodes, but then the
+# livenodes_update() thread periodically replaces this list by an up-to-date
+# list retrieved from makeing a "localnodes" requests to one of these nodes.
+livenodes = [sys.argv[1]]
+alternator_port = sys.argv[2]
+
+
+def livenodes_update():
+    global alternator_port
+    global livenodes
+    while True:
+        # Contact one of the already known nodes by random, to fetch a new
+        # list of known nodes.
+        ip = random.choice(livenodes)
+        url = 'http://{}:{}/localnodes'.format(ip, alternator_port)
+        print('updating livenodes from {}'.format(url))
+        try:
+            nodes = urllib.request.urlopen(url, None, 1.0).read().decode('ascii')
+            a = [x.strip('"').rstrip('"') for x in nodes.strip('[').rstrip(']').split(',')]
+            # If we're successful, replace livenodes by the new list
+            livenodes = a
+            print(livenodes)
+        except:
+            # TODO: contacting this ip was unsuccessful, maybe we should
+            # remove it from the list of live nodes.
+            pass
+        time.sleep(1)
+
+
+_thread.start_new_thread(livenodes_update, ())
+
+
+class Resolver:
+    def __init__(self):
+        self.address = '8.8.8.8'
```

Comment:
Nice. I see you added this feature I was meaning to add.

Reply:
I did, but the thing is it's still not working.
i.e. the dns server is working, and I can get the name `alternator` resolved.

but the damn java/YCSB doesn't seem to honor the `networkaddress.cache.ttl=4` I've set in `$JAVA_HOME/jre/lib/security/java.security`, and doesn't query the dns again.